tarsier/tag_utils.ts

```diff
@@ -305,6 +306,31 @@ function absolutelyPositionMissingTags() {
       parent.removeChild(tag);
       document.body.appendChild(tag);
     }
+
+    tags.forEach((otherTag) => {
+      if (tag === otherTag) return;
+      let otherTagRect = otherTag.getBoundingClientRect();
+
+      // reduce font of this tag and other tag until they don't overlap
+      let fontSize = parseFloat(window.getComputedStyle(tag).fontSize.split("px")[0]);
+      let otherFontSize = parseFloat(window.getComputedStyle(otherTag).fontSize.split("px")[0]);
+
+      while (
+        (tagRect.left < otherTagRect.right &&
+          tagRect.right > otherTagRect.left) &&
+        (tagRect.top < otherTagRect.bottom &&
+          tagRect.bottom > otherTagRect.top) &&
+        fontSize > 0.5 && otherFontSize > 0.5
+      ) {
+        fontSize -= 0.5;
+        otherFontSize -= 0.5;
```

Comment:
Tiny font is effectively invisible for OCR. This might have some very strange performance implications

We should also have some minimum font size tags must adhere to. Right now if aligned a certain way, couldn't two tags just go to 0px? Currently in this case, one would just be aligned on top of the other so the OCR system will get *some* tag representing the area

Reply:
True, will adjust min font size to 7px.